packages/components/src/composite/legacy.tsx

```diff
@@ -0,0 +1,301 @@
+ * all the keyboard navigation mechanisms to ensure that there's only one
+ * tab stop for the whole Composite element. This means that it can behave as
+ * a roving tabindex or aria-activedescendant container.
+ *
+ * @see https://ariakit.org/components/composite
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useMemo, useRef, useState } from '@wordpress/element';
+import deprecated from '@wordpress/deprecated';
+
+/**
+ * Internal dependencies
+ */
+import * as Current from './v2';
+
+type CompositeStore = ReturnType< typeof Current.useCompositeStore >;
+type CompositeStoreProps = NonNullable<
+	Parameters< typeof Current.useCompositeStore >[ 0 ]
+>;
+
+type Orientation = 'horizontal' | 'vertical';
+
+interface IdState {
+	baseId: string;
+	setBaseId: React.Dispatch< React.SetStateAction< IdState[ 'baseId' ] > >;
+}
+
+export interface InitialState {
+	baseId?: string;
+	unstable_virtual?: boolean;
+	rtl?: boolean;
+	orientation?: Orientation;
+	currentId?: string;
+	loop?: boolean | Orientation;
+	wrap?: boolean | Orientation;
+	shift?: boolean;
+}
+
+export interface CompositeState {
+	store: CompositeStore;
+
+	up: () => void;
+	down: () => void;
+	first: () => void;
+	last: () => void;
+	previous: () => void;
+	next: () => void;
+	move: ( id: string | undefined ) => void;
+
+	baseId: IdState[ 'baseId' ];
+	setBaseId: ( v: CompositeState[ 'baseId' ] ) => void;
+	currentId: string | undefined;
+	setCurrentId: ( v: CompositeState[ 'currentId' ] ) => void;
+	orientation: Orientation | undefined;
+	setOrientation: ( v: CompositeState[ 'orientation' ] ) => void;
+	rtl: boolean;
+	setRTL: ( v: CompositeState[ 'rtl' ] ) => void;
+	loop: boolean | Orientation;
+	setLoop: ( v: CompositeState[ 'loop' ] ) => void;
+	shift: boolean;
+	setShift: ( v: CompositeState[ 'shift' ] ) => void;
+	wrap: boolean | Orientation;
+	setWrap: ( v: CompositeState[ 'wrap' ] ) => void;
+}
+
+type FilteredCompositeStateProps = Pick< CompositeState, 'store' | 'baseId' >;
+
+type PartialCompositeState = Partial<
+	Omit< CompositeState, keyof FilteredCompositeStateProps >
+> &
+	FilteredCompositeStateProps;
+
+export type CompositeStateProps =
+	| { state: PartialCompositeState }
+	| ( PartialCompositeState & { state?: never } );
+
+interface IdState {
+	baseId: string;
+	setBaseId: React.Dispatch< React.SetStateAction< string > >;
+}
+
+function showDeprecationMessage( ...props: Parameters< typeof deprecated > ) {
+	if ( 'test' !== process.env.NODE_ENV ) {
+		deprecated( ...props );
+	}
+}
+
+const idMap = new Map< string, React.MutableRefObject< number > >();
+
+function useBaseId(
+	initialState: Pick< Partial< IdState >, 'baseId' > = {}
+): IdState {
+	const { baseId: initialBaseId } = initialState;
+	const [ baseId, setBaseId ] = useState(
+		() =>
+			initialBaseId ||
+			// eslint-disable-next-line no-restricted-syntax
+			`id-${ Math.random().toString( 32 ).substring( 2, 8 ) }`
```

Comment:
It matches the current Reakit implementation 🤷

Reply:
It's safe, and likely even better, to use `useId` if `@wordpress/components` doesn't have to support React versions earlier than or equal to 17. Reakit doesn't use it because it doesn't officially support React 18, even though most features function properly.

Reply:
I've switched to `useId`.